unit_test/dependency_constraints_test.py

```diff
@@ -10,7 +10,8 @@ def test_defaults():
     resources_dir = project_root / 'cibuildwheel' / 'resources'
 
     assert dependency_constraints.base_file_path.samefile(resources_dir / 'constraints.txt')
-    assert dependency_constraints.get_for_python_version('3.8').samefile(resources_dir / 'constraints.txt')
+    assert dependency_constraints.get_for_python_version('3.99').samefile(resources_dir / 'constraints.txt')
```

Comment:
Is there a reason to still have a default `constraints.txt` file?

I thought the reason it's there is because 3.8 and 3.9 currently have the exact same version file?

Reply:
see @joerick comments https://github.com/joerick/cibuildwheel/pull/490#discussion_r546425734 on his review.

Reply:
Yeah, I saw, but to start customizing your own constraints, you could also start from `constraints-python39.txt`? `constraints.txt` feels like a historical remainder, somehow. But ofc, it's not wrong; I was just thinking if it's not better to get a reminder (in the form of a failing CI), when we add 3.10, and future versions to `cibuildwheel`?

Reply:
I think it's simpler just to have one per version. "But just copying the latest Python we support should do" I think is exactly what @YannickJadoul is referring to here.

Reply:
Yeah, we're coming from a situation at the start where we only had one `constraints.txt`. Then 2.7 started acting up, and we had `constraints-python2.7.txt`, then 3.5, then 3.6, ... and now, there seems to be the realization that just having a separate file for all is better.

So why not just ditch the "default", then, if falling back to the default is a mistake?

Reply:
Not that crucial, though. Just wondering if there's a good reason. (There's nothing else to nitpick about, so I need to find _something_? I very much like the idea and rest of the PR :-) )

Reply:
@YannickJadoul, @henryiii, I agree and proposed to do so. Final decision left to @joerick.

Reply:
Right. Sorry; should have added my comment to that thread, then :-)

Reply:
I'm happy to go with that - that we don't bundle a constraints.txt. I think as @YannickJadoul says, the version-specific ones appeared to solve specific problems, but version-specific is probably a more robust solution. We'll have to remove [this line](https://github.com/joerick/cibuildwheel/blob/master/cibuildwheel/util.py#L98) but that's not a huge deal, I think.

Reply:
Hmmm, looking at that code, I'm now seeing it's because we want to keep supporting a default `constraints.txt` for users.

I'm fine either way, then! As you see fit, @mayeut. Sorry about the confusion!